CIScripts/Test/Tests/Pkt0.Tests.ps1

```diff
@@ -0,0 +1,81 @@
+Param (
+    [Parameter(Mandatory=$true)] [string] $TestbedAddr
+)
+
+. $PSScriptRoot\..\Utils\CommonTestCode.ps1
+. $PSScriptRoot\..\TestConfigurationUtils.ps1
+. $PSScriptRoot\..\..\Common\Aliases.ps1
+. $PSScriptRoot\..\..\Common\VMUtils.ps1
+
+. $PSScriptRoot\..\GetTestConfigurationCodiLegacy.ps1
+$TestConf = Get-TestConfiguration
+$Session = New-PSSession -ComputerName $TestbedAddr -Credential (Get-VMCreds)
+
+Describe "vRouter Agent" {
+    Context "given vRouter Forwarding Extension is NOT running" {
+        It "crashes" {
+            Assert-AgentProcessCrashed -Session $Session -After $TimeBeforeCrash | Should Throw
+        }
+
+        BeforeEach {
+            Disable-VRouterExtension -Session $Session `
+                -AdapterName $TestConf.AdapterName `
+                -VMSwitchName $TestConf.VMSwitchName `
+                -ForwardingExtensionName $TestConf.ForwardingExtensionName
+            Test-IsVRouterExtensionEnabled -Session $Session `
+                -VMSwitchName $TestConf.VMSwitchName `
+                -ForwardingExtensionName $TestConf.ForwardingExtensionName `
+                | Should Be $false
+            $TimeBeforeCrash = Invoke-Command -Session $Session -ScriptBlock { Get-Date }
+            Enable-AgentService -Session $Session
+        }
+    }
+
+    Context "given vRouter Forwarding Extension is running" {
+        It "runs correctly" {
+            $Result = Assert-IsAgentServiceEnabled -Session $Session
+            $Result | Should BeNullOrEmpty
```

Comment:
`Should BeNullOrEmpty` is unnecessary -- `Assert-IsAgentServiceEnabled` either returns null or throws. (the return inside of it is inside a ScriptBlock and can be removed.